src/EditorFeatures/CSharpTest/Formatting/CSharpNewDocumentFormattingServiceTests.cs

```diff
@@ -15,8 +17,98 @@ namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.Formatting
     public class CSharpNewDocumentFormattingServiceTests : AbstractNewDocumentFormattingServiceTests
     {
         protected override string Language => LanguageNames.CSharp;
-        protected override TestWorkspace CreateTestWorkspace(string testCode)
-            => TestWorkspace.CreateCSharp(testCode);
+        protected override TestWorkspace CreateTestWorkspace(string testCode, ParseOptions? parseOptions)
+            => TestWorkspace.CreateCSharp(testCode, parseOptions);
+
+        [Fact]
+        public async Task TestFileScopedNamespaces()
+        {
+            await TestAsync(testCode: @"
+namespace Goo
+{
+    internal class C
+    {
+    }
+}",
+            expected: @"
+namespace Goo;
+
+internal class C
+{
+}
+",
+            options: new[]
+            {
+                (CSharpCodeStyleOptions.NamespaceDeclarations, new CodeStyleOption2<NamespaceDeclarationPreference>(NamespaceDeclarationPreference.FileScoped, NotificationOption2.Error))
+            },
+            parseOptions: new CSharpParseOptions(LanguageVersion.CSharp10));
+        }
+
+        [Fact]
+        public async Task TestFileScopedNamespaces_Invalid1()
```

Comment:
Invalid{1, 2} could use an explanation of _why_ they are invalid. It took me a second to realize 2 was because of the language version.

Reply:
Renamed